### PR TITLE
revert!: multi share commitment

### PR DESCRIPTION
**Pull request by rootulp**
Closes https://github.com/celestiaorg/celestia-app/issues/1231

### CODE REVIEW

File: x/blob/types/payforblob.go [hunk 1]
```diff
@@ -61,10 +61,10 @@ func (msg *MsgPayForBlob) Type() string {
 // ValidateBasic fulfills the sdk.Msg interface by performing stateless
 // validity checks on the msg that also don't require having the actual blob
 func (msg *MsgPayForBlob) ValidateBasic() error {
-	if len(msg.NamespaceIds) != len(msg.ShareVersions) || len(msg.NamespaceIds) != len(msg.BlobSizes) {
+	if len(msg.NamespaceIds) != len(msg.ShareVersions) || len(msg.NamespaceIds) != len(msg.BlobSizes) || len(msg.NamespaceIds) != len(msg.ShareCommitments) {
```
Comment (cmwaters) [inline x/blob/types/payforblob.go:64]: Do we also want to add a check that this length is not zero?

**Review by cmwaters — APPROVED** [review]
lgtm

File: app/estimate_square_size_test.go [hunk 2]
```diff
@@ -88,7 +88,7 @@ func Test_estimateSquareSize_MultiBlob(t *testing.T) {
 			"100 multiblob single share transaction", func() [][]int {
 				return [][]int{blobfactory.Repeat(int(100), 100)}
 			},
-			16, 5,
+			16, 11,
```
Comment (evan-forbes) [inline app/estimate_square_size_test.go:91]: [simple comment]
perhaps we can eventually include some some option to use a multiblob commit and no namespaces to be more efficient in the future

**Review by evan-forbes — APPROVED** [review]
makes sense, LGTM :+1: :+1: other than Callum's comment

File: x/blob/types/payforblob.go (hunk 1 above)
Comment (rootulp) [reply x/blob/types/payforblob.go:64]: By this length are you referring to msg.ShareCommitments or msg.NamespaceIds? So something like this:

```go
	if len(msg.NamespaceIds) == 0 {
		return ErrNoNamespaceIds
	}
```

or 

```go
	if len(msg.ShareCommitments) == 0 {
		return ErrNoShareCommitments
	}
```

Comment (rootulp) [reply x/blob/types/payforblob.go:64]: IMO both would work. We could include checks that all of these fields are non-zero: NamespaceIds, ShareVersions, BlobSizes, and ShareCommitments. Will do in a follow-up PR.

**Pull request merged**